src/platforms/esp32/main/network_driver.c

```diff
@@ -132,6 +133,28 @@ void network_driver_start(Context *ctx, term_ref pid, term_ref ref, term config)
         ESP_LOGI("NETWORK", "starting wifi: SSID: [%s], password: [XXXXXXXX].", wifi_config.sta.ssid);
         ESP_ERROR_CHECK(esp_wifi_start());
 
+        char dhcp_hostname[TCPIP_HOSTNAME_MAX_SIZE + 1];
+        if (!term_is_nil(dhcp_hostname_value)) {
+            int ok = 0;
+            char *tmp = interop_term_to_string(dhcp_hostname_value, &ok);
+            if (!ok || IS_NULL_PTR(tmp)) {
+                term reply = port_create_error_tuple(ctx, BADARG_ATOM);
+                port_send_reply(ctx, pid, ref, reply);
+                return;
+            } else {
+                strncpy(dhcp_hostname, tmp, TCPIP_HOSTNAME_MAX_SIZE);
+                dhcp_hostname[TCPIP_HOSTNAME_MAX_SIZE] = '\0';
```

Comment:
no need for this line, strncpy always add \0 at the end.

Reply:
Not if dhcp_hostname is of length TCPIP_HOSTNAME_MAX_SIZE or greater.

Per the C documentation:

    "The stpncpy() and strncpy() functions copy at most len characters from
     src into dst.  If src is less than len characters long, the remainder of
     dst is filled with `\0' characters.  Otherwise, dst is not terminated."

Reply:
I had considered checking the length and only putting the null terminator if the dhcp_hostname was too long, but figured it would be more efficient to just tack the null terminator on, regardless.  I agree it's an extra write, but less than a scan and then a potential write.  Happy to do it either way, but I think we do need to ensure the string is null terminated, in case it's too long.